packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -0,0 +1,648 @@
+    super.initState();
+    _autocompleteController = widget.autocompleteController
+        ?? AutocompleteController<T>(options: widget.options);
+    _listenToController(_autocompleteController);
+
+    SchedulerBinding.instance.addPostFrameCallback((Duration _) {
+      _updateOverlay();
+    });
+  }
+
+  @override
+  void didUpdateWidget(AutocompleteCore<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    // Handle changes in the AutocompleteController.
+    if (widget.autocompleteController == null && oldWidget.autocompleteController != null) {
+      _unlistenToController(oldWidget.autocompleteController);
+      _autocompleteController = AutocompleteController<T>(
+        options: widget.options,
+      );
+      _listenToController(_autocompleteController);
+    } else if (widget.autocompleteController != null && oldWidget.autocompleteController == null) {
+      _unlistenToController(_autocompleteController);
+      _autocompleteController.dispose();
+      _autocompleteController = widget.autocompleteController;
+      _listenToController(_autocompleteController);
+    } else if (widget.autocompleteController != oldWidget.autocompleteController) {
+      _unlistenToController(oldWidget.autocompleteController);
+      _autocompleteController = widget.autocompleteController;
+      _listenToController(_autocompleteController);
+    }
+
+    SchedulerBinding.instance.addPostFrameCallback((Duration _) {
+      _updateOverlay();
+    });
+  }
+
+  @override
+  void dispose() {
+    _unlistenToController(_autocompleteController);
+    if (widget.autocompleteController == null) {
+      _autocompleteController.dispose();
+    }
+    _floatingResults?.remove();
+    _floatingResults = null;
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      key: _fieldKey,
+      child: CompositedTransformTarget(
+        link: _resultsLayerLink,
+        child: widget.buildField(
+          context,
+          _autocompleteController.textEditingController,
+          _onFieldSubmitted,
+        ),
+      ),
+    );
+  }
+}
+
+class _FloatingResults<T> extends StatelessWidget {
+  const _FloatingResults({
+    Key key,
+    @required this.buildResults,
+    @required this.fieldSize,
+    @required this.layerLink,
+    @required this.onSelected,
+    @required this.results,
+  }) : assert(buildResults != null),
+       assert(fieldSize != null),
+       assert(layerLink != null),
+       assert(onSelected != null),
+       assert(results != null),
+       super(key: key);
+
+  final AutocompleteResultsBuilder<T> buildResults;
+  final Size fieldSize;
+  final LayerLink layerLink;
+  final AutocompleteOnSelected<T> onSelected;
+  final List<T> results;
+
+  @override
+  Widget build(BuildContext context) {
+    return Positioned(
+      width: fieldSize.width,
+      child: CompositedTransformFollower(
+        link: layerLink,
+        showWhenUnlinked: false,
+        offset: Offset(
```

Comment:
Should the positioning change when the writing direction is RTL?

Reply:
That's a really good point, I hadn't thought much about it, but it seems to be good as-is.  The width of the results here is set to be as wide as the field, so that gives the user the space to align things how they want.  If I wrap my basic example in Directionality RTL, it works itself out fine:

<img width="377" alt="Screen Shot 2020-08-28 at 3 29 43 PM" src="https://user-images.githubusercontent.com/389558/91620008-9a5ba780-e943-11ea-9e9c-c2a6049457ea.png">

Reply:
Where should the dropdown menu go if the input field is placed close to the bottom of the screen?